### PR TITLE
KOGITO-4671 : New elements should always be connected by its central magnetic point

### PR DESCRIPTION
**Pull request by inodeman**
Hi @romartin can you review this PR? Artifact can be found here. https://drive.google.com/drive/folders/1tjkhXfTHDic8Ef4wiLLmx3pgzK70RIiQ?usp=sharing

### CODE REVIEW

**Comment by domhanak** [comment]
jenkins run fdb

**Comment by domhanak** [comment]
Yea @inodeman please rebase to get the FDB check :)

**Review by romartin — APPROVED** [review]

**Comment by romartin** [comment]
@domhanak artifacts generated, ready to test once possible :) Thanks in advance!

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/canvas/controls/BPMNCreateNodeActionTest.java [hunk 1]
```diff
@@ -55,15 +68,50 @@ public void setUp() {
...
+        assertTrue(isGatewaySourceOrTarget(gatewayNode, gatewayNode));
+        assertFalse(isGatewaySourceOrTarget(taskNode, taskNode));
+        assertFalse(isGatewaySourceOrTarget(eventNode, taskNode));
+        assertFalse(isGatewaySourceOrTarget(taskNode, eventNode));
+        assertFalse(isGatewaySourceOrTarget(eventNode, eventNode));
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testbuildConnectionBetween() {
```
Comment (domhanak) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/canvas/controls/BPMNCreateNodeActionTest.java:85]: could we split this into multiple test cases?

**Comment by domhanak** [comment]
@inodeman @romartin LGTM, could you please report this issue you mention in comments:
```Current implemenetation is set for gateays to have a fixed magenet from source and to target, if changed to center it actually does not look right as there is a space gap, so leaving new nodes to be connected to center, except Gateways.```
I think that this is a bug and should be fixed in the future & as this fix is also useful for gateways.

**Review by domhanak — APPROVED** [review]
Approving, please split the unit tests into units focused on one case per test.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/canvas/controls/BPMNCreateNodeActionTest.java (hunk 1 above)
Comment (inodeman) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/canvas/controls/BPMNCreateNodeActionTest.java]: Hi @domhanak separated tests now :-)

**Comment by domhanak** [comment]
@inodeman The test failed in FDB is unstable currently and we are investigating so good to merge anyway.

**Comment by domhanak** [comment]
@inodeman @romartin https://github.com/kiegroup/kogito-editors-java/pull/18#issuecomment-845147218 see my question please.

**Comment by inodeman** [comment]
Hi @domhanak say your comment, so you mean that in the future all nodes should be connected from the center even gateways after the gap is fixed? Let me know

**Comment by romartin** [comment]
Hey @domhanak  @inodeman 

Not really why you talk about some issue guys sorry, in the comment https://github.com/kiegroup/kogito-editors-java/pull/18#issuecomment-845147218 I see no issue, nothing to be reported. Consider the behavior for magnets in gateways has not being changed, we still have the same behavior as before this work. This work applies to all other nodes, but not for gateways. In fact we are using fixed magnets for gateways intentionally, it was a requirement being asked by Kris long time ago, and this work stays with same approach. Does it makes sense?

Sooo can I go ahead with merging guys?

**Comment by domhanak** [comment]
@inodeman @romartin I am talking about this GAP that is emerging see the picture:
gap-connection
I think that it is a bug, is this reported somewhere? I remember this was discussed some time ago and I can't find any JIRA :(

**Comment by domhanak** [comment]
> Hi @domhanak say your comment, so you mean that in the future all nodes should be connected from the center even gateways after the gap is fixed? Let me know

Yes that is what we  eventually want anyway for our editors. I am curious about the GAP issues, is it reported? Could you please confirm?

**Comment by domhanak** [comment]
Found it, I was looking in KOGITO jira not JBPM :confused: GAP issue reported here -  https://issues.redhat.com/browse/KOGITO-5210 

Let's merge.

**Comment by domhanak** [comment]
The test failure in FDB of VSCode is unrelated.

**Pull request merged**